rest/blip_sync.go

```diff
@@ -66,17 +67,22 @@ func userBlipHandler(underlyingMethod blipHandlerMethod) blipHandlerMethod {
 
 	wrappedBlipHandler := func(bh *blipHandler, bm *blip.Message) error {
 
-		oldUser := bh.db.User()
-		if oldUser == nil {
-			return fmt.Errorf("nil user for blip handler")
+		var newUser auth.User
+		if oldUser := bh.db.User(); oldUser != nil {
+			var err error
+			newUser, err = bh.db.Authenticator().GetUser(oldUser.Name())
+			if err != nil {
+				return err
+			}
 		}
 
-		// Create a new user-scoped database on each blip request (otherwise runs into SG issue #2717)
-		newUser, err := bh.db.Authenticator().GetUser(oldUser.Name())
+		// Create a user-scoped database on each blip request (otherwise runs into SG issue #2717)
```

Comment:
I think we can take everything from line 79 to 87 and stick inside the if clause from line 71 (and move line 70 inside while we're at it).  If the user is nil, we don't need to touch the existing bh.database (and so also don't need to worry about read/write races).

Reply:
Ah right, of course. That makes sense, thanks.